posthog/temporal/data_imports/pipelines/pipeline/utils.py

```diff
@@ -259,6 +262,61 @@ def _append_debug_column_to_pyarrows_table(table: pa.Table, load_id: int) -> pa.
     return table.append_column("_ph_debug", column)
 
 
+def should_partition_table(delta_table: deltalake.DeltaTable | None, schema: ExternalDataSchema) -> bool:
+    if not schema.is_incremental:
+        return False
+
+    if delta_table is None:
+        return True
+
+    delta_schema = delta_table.schema().to_pyarrow()
+    if PARTITION_KEY in delta_schema.names:
+        return True
+
+    return False
+
+
+def append_partition_key_to_table(
+    table: pa.Table, incremental_field: str, incremental_field_type: IncrementalFieldType, logger: FilteringBoundLogger
+) -> pa.Table:
+    if (
+        incremental_field_type != IncrementalFieldType.Date
+        and incremental_field_type != IncrementalFieldType.DateTime
+        and incremental_field_type != IncrementalFieldType.Timestamp
+    ):
+        logger.debug(f"No partition key added due to incremental_field_type={incremental_field_type}")
+        return table
+
+    partition_array: list[str | None] = []
+
+    for value in table.column(incremental_field):
```

Comment:
This could be enhanced to use pyarrow.compute functions: https://arrow.apache.org/docs/python/api/compute.html. This way, we would save the memory from moving the value from an Arrow array to a Python object. I imagine there should be a solution that involves the pyarrow compute functions `pyarrow.compute.year_month_day` or something similar.

Anyways, this could be enhanced later.